sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -237,15 +237,32 @@ static unsigned int generic_calc_stream_format(struct snd_sof_dev *sdev,
 					       struct snd_pcm_substream *substream,
 					       struct snd_pcm_hw_params *params)
 {
+	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
```

Comment:
typo in commit message:
capture introduces a **different** between the channels handled

and this reads a bit odd:
We need to reconstruct the number of channels to figure out how many channels, ...

Reply:
fixed and reworded completely, please recheck.